### ISSUE TITLE
new a11y rule: mouse-events-have-key-events

### ISSUE THREAD

**Issue by HamletDRC** [Microsoft/tslint-microsoft-contrib#279]
mouse-events-have-key-events

Enforce that onMouseOver/onMouseOut are accompanied by onFocus/onBlur for keyboard-only users.
    • http://oaa-accessibility.org/wcag20/rule/59/
    • http://oaa-accessibility.org/wcag20/rule/60/

onmousedown, onmouseup and onmousemove event handlers should have keyboard equivalents.
    • http://oaa-accessibility.org/wcag20/rule/63/

Check that elements with mouse event handlers also have key event handlers
    • http://oaa-accessibility.org/wcag20/rule/95/

Check that elements with ondblclick handles also have key event handlers
    • http://www.w3.org/TR/WCAG20-TECHS/SCR20
    • From HTML Code Sniffer (https://squizlabs.github.io/HTML_CodeSniffer/Standards/WCAG2/)

**Comment by lizzzp1** [comment]
Would like this one as well. I can take this. Anything additional to note than what the description supplies?

**Comment by JoshuaKGoldberg** [comment]
Awesome, thanks @lizzzp1! There's not much that I can think of. One corner case would be `...` props: the rules should use the type checker to determine if the handlers are in them.

**Comment by lizzzp1** [comment]
@JoshuaKGoldberg do you think these are one rule or two different ones?
Based on peeking at this:
https://github.com/reactjs/react-a11y/blob/master/docs/rules/click-events-have-key-events.md
https://github.com/reactjs/react-a11y/blob/master/docs/rules/mouse-events-have-key-events.md

**Comment by JoshuaKGoldberg** [comment]
@lizzzp1 +1 for the prior art. Yes, let's make these two different rules.

**Comment by JoshuaKGoldberg** [comment]
Hmm, now I'm wondering whether it makes sense to take on the overhead of a typed rule... let's say that can be an optional nice-to-have as a followup.
